src/pipelines/data_gen.py

```diff
@@ -194,8 +194,11 @@ def w2v(doc, corpus, w2v_model, w2v, feature):
 
     if w2v.get('avg', False):
         docw2v = run_w2v(w2v_model, doc, w2v)
-        corpusw2v = run_w2v(w2v_model, corpus, w2v)
-        vectors = [docw2v, corpusw2v]
+        background_vectors = list()
+        for item in background_docs:
+            background_vectors.append(run_w2v(w2v_model, item, w2v))
+        backgroundw2v = np.mean(background_vectors, axis=0)
```

Comment:
Yes, this makes a lot of sense. I was originally reluctant to only report the extreme values from a large potential pool of vectors but then again the average probably weakens the signal. I'll put in the change.